docs/root/development/performance/binary_size.rst

```diff
@@ -0,0 +1,164 @@
+.. _dev_performance_size:
+
+Analysis of Binary Size
+=======================
+
+In order to be able to tackle binary size analysis,
+the Envoy Mobile team has standardized the process in this document.
+
+Object files analysis
+---------------------
+
+Getting a binary
+~~~~~~~~~~~~~~~~
+
+In order to have consistency of results this is the toolchain used to build the
+binary for analysis:
+
+1. clang-8
+2. lld (installed with clang)
+3. arm64 machine
+
+The arm64 machine used to build the test binary was an arm64 ec2 a1.2xlarge
+instance with Ubuntu 18.04. The following script was run to set up the
+necessary tools::
+
+  # basic toolchain
+  sudo apt-get install build-essential openjdk-8-jdk python zip unzip \
+    software-properties-common make cmake bc libtool ninja-build automake \
+    time apt-transport-https
+  # clang-8
+  wget http://releases.llvm.org/8.0.0/clang+llvm-8.0.0-aarch64-linux-gnu.tar.xz
+  sudo tar -C /usr/local/ -xvf clang+llvm-8.0.0-aarch64-linux-gnu.tar.xz --strip 1
+  rm -rf clang+llvm-8.0.0-aarch64-linux-gnu.tar.xz
+  # bazel 0.26.1
+  wget https://github.com/bazelbuild/bazel/releases/download/0.26.1/bazel-0.26.1-dist.zip
+  mkdir -p /tmp/bazel_build
+  unzip -o bazel-0.26.1-dist.zip -d /tmp/bazel_build
+  rm -rf bazel-0.26.1-dist.zip
+  cd /tmp/bazel_build
+  env EXTRA_BAZEL_ARGS="--host_javabase=@local_jdk//:jdk" bash ./compile.sh
+  sudo cp output/bazel /usr/local/bin/bazel
+  rm -rf /tmp/bazel_build
+  # bloaty
+  git clone https://github.com/google/bloaty
+  cd bloaty
+  mkdir build
+  cd build
+  cmake ..
+  make -j6
+  cp bloaty /usr/local/bin/bloaty
+
+The binary being compiled is ``//test/performance:test_binary_size``.
+The binary is getting built with the following build command::
+
+  bazel build //test/performance:test_binary_size --config=sizeopt --copt=-ggdb3 --linkopt=-fuse-ld=lld
```

Comment:
It's a bit unclear to me from reading this how the flags below map to these configuration flags

Reply:
clarified below.